langfuse/_client/attributes.py

```diff
@@ -61,6 +64,16 @@ class LangfuseOtelSpanAttributes:
     AS_ROOT = "langfuse.internal.as_root"
 
 
+def propagate_attributes(
+    *,
+    current_ctx: otel_context.Context,
+    dict_to_propagate: Dict[str, Any],
+) -> otel_context.Context:
+    for key, value in dict_to_propagate.items():
+        current_ctx = baggage.set_baggage(key, str(value), context=current_ctx)
+    return current_ctx
```

Comment:
The context is immutable, i.e. every set_baggage operation produces a new, updated context with the relevant properties. We update whatever context we get and return the new one.